### PR TITLE
[PATCH v2] Merge ODP linux-generic v1.29.0.0 into linux-dpdk

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

File: scripts/ci/build.sh [hunk 1]
```diff
@@ -28,11 +28,15 @@ ODP_LIB_NAME=libodp-dpdk
 fi
 
 # Additional warning checks
-EXTRA_CHECKS="-Werror -Wall -Wextra -Wconversion -Wundef -Wpointer-arith -Wfloat-equal -Wpacked"
+EXTRA_CHECKS="-Werror -Wall -Wextra -Wconversion -Wfloat-equal -Wpacked"
```
Comment (JereLeppanen) [inline scripts/ci/build.sh:31]: > Also, remove checks already included in ODP library build (undef, pointer-arith).

But shouldn't we have here some pretty strict set of warning options that an application could use, regardless of what has been used while building the libraries?

Comment (MatiasElo) [reply scripts/ci/build.sh:31]: The same warning options (set with `ODP_CHECK_CFLAG`) are also used when building the example applications during normal `make` (the commit message could be better). My reasoning for removing these warnings from here was just to make the list shorter and easier to maintain.

Comment (JereLeppanen) [reply scripts/ci/build.sh:31]: Ok, sounds reasonable, at least for now.

**Review by JereLeppanen — APPROVED** [review]

**Comment by MatiasElo** [comment]
Thanks for the review!

**Pull request merged**